databases/business-sync/fragment-shoebox-item.js

```diff
@@ -0,0 +1,39 @@
+{
+  channels: getDocSyncChannels(doc, oldDoc),
+  typeFilter: function(doc, oldDoc) {
+    // Keyspace schema:  biz.<biz_id>.shoeboxItem.<item_type>.<uuid>
+    return createBusinessEntityRegex('shoeboxItem\\.[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]+').test(doc._id);
+  },
+  allowUnknownProperties: true,
+  immutable: true,
+  propertyValidators: {
+    type: {
+      // The type of shoebox item (ie, bank transaction, email, photo, digital receipt)
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    source: {
+      // The source of the item
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    sourceId: {
```

Comment:
What is the use case for having the source id surfaced here? I'm not sure it will be helpful, would like to chat it through.

Reply:
Basically to support tracing it back.  A redundancy measure in the rare case that a received entity does not contain the ID as a field (CB documents are like this).  You had previously given your support to such a field:
> +1 on adding a source identifier.